eval/helm/main.py

```diff
@@ -0,0 +1,185 @@
+# Credits: https://github.com/llm-efficiency-challenge/neurips_llm_efficiency_challenge
+import json
+import logging
+import sys
+import time
+import warnings
+from pathlib import Path
+from typing import Dict, List, Literal, Optional
+
+import lightning as L
+import torch
+import uvicorn
+from fastapi import FastAPI
+from lightning.fabric.strategies import FSDPStrategy
+from pydantic import BaseModel
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt import GPT, Config, Tokenizer
+from lit_gpt.model import Block
+from lit_gpt.utils import check_valid_checkpoint_dir, lazy_load, quantization
+
+from generate import generate
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+app = FastAPI()
+
+
+class ProcessRequest(BaseModel):
+    prompt: str
+    num_samples: int = 1
+    max_new_tokens: int = 50
+    top_k: int = 200
+    temperature: float = 0.8
+    seed: Optional[int] = None
+    echo_prompt: Optional[bool]
+
+
+class Token(BaseModel):
+    text: str
+    logprob: float
+    top_logprob: Dict[str, float]
+
+
+class ProcessResponse(BaseModel):
+    text: str
+    tokens: List[Token]
+    logprob: float
+    request_time: float
+
+
+class TokenizeRequest(BaseModel):
+    text: str
+    truncation: bool = True
+    max_length: int = 2048
+
+
+class TokenizeResponse(BaseModel):
+    tokens: List[int]
+    request_time: float
+
+
+def main(
+    checkpoint_dir: str = "",
+    precision: str = "bf16-true",
+    device="auto",
+    devices: int = 1,
+    strategy: str = "auto",
+    quantize: Optional[
+        Literal[
+            "bnb.nf4", "bnb.nf4-dq", "bnb.fp4", "bnb.fp4-dq", "bnb.int8", "gptq.int4"
```

Comment:
bnb.int8 won't work because it requires a different weights path. You can see other scripts as examples or just remove the option here